### DISCUSSION TITLE
anonfile 1.0.0

### DISCUSSION THREAD

**Discussion by nstrydom2** [nstrydom2/anonfile-api#94] [Releases]
We are happy to announce the release of anonfile 1.0.0 🎉 

This release includes updated dependencies and incorporates more robust testing. Check out our [changlog](https://github.com/nstrydom2/anonfile-api/blob/master/CHANGELOG.md) for a more detailed list of changes. We recommend users upgrade to this version.

Please report any issues with the release on the [anonfile issue tracker](https://github.com/nstrydom2/anonfile-api/issues).

<hr /><em>This discussion was created from the release <a href='https://github.com/nstrydom2/anonfile-api/releases/tag/v1.0.0'>anonfile 1.0.0</a>.</em>

**Comment by StefanGreve** [comment]
Regarding the recent build failures, I have looked a little bit into what's happening in the log files on GitHub Actions. Firstly, it is pretty clear that the error stems from the `python -m build` command ([log](https://github.com/nstrydom2/anonfile-api/actions/runs/5746044350/job/15574933773#step:5:66)).

In PR #95 I added the `MANIFEST.in` file hoping this would fix our problem. While that evidently doesn't seem to be the case, I still believe that this was at least a partial solution.

There is a closed issue on the `build` repository (https://github.com/pypa/build/issues/427) that looks pretty similar to what we are encountering here. In the preceding issue (https://github.com/pypa/build/issues/426) there's a comment that suggested the following fix:

```cli
python -m build --no-isolation
```

Apparently it takes some playing around to get the tooling to work correctly.

**Comment by StefanGreve** [comment]
@nstrydom2 I just remembered that rebuilding the job won't do anything good, because it will pull version `1.0.0` for the build since the current triggers are set to `[published, edited]`. In other words, to test the build fixes you need to create a new release with the tag `v1.0.1` (which is actually in tune with what semantic versioning prescribes for this situation). I already went ahead and pushed this tag on `master`, but you will need to create a new release.

~~If this doesn't work out, we will need to retract release `1.0.1`, delete tag `v1.0.1` and repeat until we find a solution, though I am fairly confident that the next release should do the trick.~~ When I run `python -m build` locally I get a successful build with and without the `--no-isolation` flag. 

I added the `workflow_dispatch` event to the `python-publish.yml` workflow file as well. Version `1.0.0` is out now!

Obviously this situation was not ideal, but I have a few ideas how we can improve this workflow during development so that the actual release build doesn't need to be experimented with :)

**Update**: There's a feature proposal about this online now: #98
